crates/pathfinder/src/update.rs

```diff
@@ -0,0 +1,170 @@
+//! Discover pathfinder releases via github API.
+
+/// Monitors Github for new releases and logs when this occurs.
+///
+/// Will continuously log this with every poll so that the user
+/// has a better chance of spotting it.
+pub async fn poll_github_for_releases() -> anyhow::Result<()> {
+    use anyhow::Context;
+    let current_version = env!("VERGEN_GIT_SEMVER_LIGHTWEIGHT");
+    let current_version = current_version.strip_prefix('v').unwrap_or(current_version);
+    let local_version = semver::Version::parse(current_version)
+        .context("Semver parsing of local version failed")?;
+    let mut latest_gh_version = None;
+
+    let client = reqwest::Client::new();
+    let mut etag = None;
+
+    loop {
+        match fetch_latest_github_release(&client, &etag).await {
+            UpdateResult::Update(release) => {
+                etag = release.etag;
+
+                let release_version = release
+                    .version
+                    .strip_prefix('v')
+                    .unwrap_or(&release.version);
+                match semver::Version::parse(release_version) {
+                    Ok(version) => {
+                        latest_gh_version = Some(version);
+                    }
+                    Err(e) => {
+                        tracing::warn!(error=%e, version=%release.version, "Semver parsing of latest github release failed")
```

Comment:
I hope this and other `warn!()`s on error won't freak out the users.

Reply:
Well we probably want to know that we have a bad semver version tag out :) We could do it more complicated by filtering by tags, but then we need to query all of the versions.

Hmm. Maybe it won't be so bad with the ETag set since it will only occur once per release update? Might be worth considering as an option to make this more robust.